WordPress/gutenberg
# Display Checkbox by default in dataviews

## What?
Fixes https://github.com/WordPress/gutenberg/issues/59175

This PR makes checkboxes in table and grid layouts always visible by removing hidden states and unnecessary hover/focus styles.

## Why?
The changes improve:

- Accessibility: Checkboxes are now always visible and discoverable.
- Usability: Users can see available actions immediately.
- Consistency: Predictable interface behavior reduces cognitive load.

## How?

- Made checkboxes always visible in both table and grid layouts.
- Removed the hidden states and hover effects that previously controlled their visibility.
- Kept other interactive styles, such as selected states, to maintain clarity.

## Testing Instructions

- Navigate to the Site Editor > Pages > Manage all pages.
- Verify that checkboxes in both table and grid layouts are always visible.
- Interact with checkboxes to ensure functionality and styling remain intact.

## Screenshots or screencast 


https://github.com/user-attachments/assets/47f947ab-21eb-421c-8bdd-31bd578b0a34
